### PR TITLE
OptiBP Widget

### PR DESCRIPTION
**Pull request by hamza-vd**
See #570

### CODE REVIEW

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/JsonFormMLSAssetGenerator.java [hunk 1]
```diff
@@ -351,7 +351,7 @@ private static JsonFormInteractor getJsonFormInteractor() throws Exception {
                 ? "com.vijay.jsonwizard.interactors.JsonFormInteractor" : jsonFormInteractorName;
         Class<?> clazz = Class.forName(jsonFormInteractorName);
         Method factoryMethod = clazz.getDeclaredMethod("getInstance");
-        return (JsonFormInteractor) factoryMethod.invoke(null, null);
+        return (JsonFormInteractor) factoryMethod.invoke(null,  (Object[]) null);
```
Comment (dubdabasoduba) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/JsonFormMLSAssetGenerator.java:354]: Any reason we are casting this to an Object?

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-VERSION_NAME=2.0.11-SNAPSHOT
+VERSION_NAME=2.1.1-SNAPSHOT
```
Comment (dubdabasoduba) [inline gradle.properties:1]: Why are we jumping to 2.1.1 instead of 2.1.0?

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/JsonFormMLSAssetGenerator.java (hunk 1 above)
Comment (hamza-vd) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/JsonFormMLSAssetGenerator.java:354]: Unit tests failing with TypeCastException for the class on this branch.

File: gradle.properties (hunk 2 above)
Comment (hamza-vd) [reply gradle.properties]: Reverted to v2.1.0

**Review by allan-on — COMMENTED** [review]
Are the `NativeRadioButtonFactory` and `CheckBoxFactory` tests specifically failing on this branch?

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/constants/JsonFormConstants.java [hunk 3]
```diff
@@ -312,6 +313,21 @@
         public static final int RC_HANDLE_GMS = 9001;
     }
 
+    public static class OptibpConstants {
+        public static final String OPTIBP_WIDGET = "optibp";
+        public static final int OPTIBP_REQUEST_CODE = 301;
+        public static final int OPTIBP_REPEAT_REQUEST_CODE = 302;
+        public static final String OPTIBP_REPORT_RESULT = "result";
+        public static final String OPTIBP_REPORT_COMPONENT = "component";
+        public static final String OPTIBP_REPORT_VALUE_QUANITITY = "valueQuantity";
```
[the remark this answers is not in the document]
Comment (hamza-vd) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/constants/JsonFormConstants.java:322]: Yes, I updated some functionality in `NativeRadioButtonFactory` and `CheckBoxFactory` to fix a `FATAL EXCEPTION` appearing on Android 7, resulting in failing tests for these 2. So after that, I also had to update tests for these.

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**